src/FirebaseAuth.js

```diff
@@ -23,9 +23,13 @@
 
   FirebaseAuth = function($firebaseUtils, auth) {
     this._utils = $firebaseUtils;
-    if (typeof ref === 'string') {
-      throw new Error('Please provide a Firebase reference instead of a URL when creating a `$firebaseAuth` object.');
+
+    if (typeof auth === 'string') {
+      throw new Error('The $firebaseAuth service accepts a Firebase auth instance (or nothing) instead of a URL.');
+    } else if (typeof auth.ref !== 'undefined') {
+      throw new Error('The $firebaseAuth service accepts a Firebase auth instance (or nothing) instead of a Database reference.');
```

Comment:
My issue is that if we throw errors mentioning things like `firebase.database.Database` and you [go to google that](https://encrypted.google.com/search?hl=en&q=firebase.database.Database) it will provide no insight or description of what that is. Even in the docs it's only discussed in the reference, so although correct, I'm not sure this is helpful.